app/controllers/results_submission_controller.rb

```diff
@@ -12,6 +12,28 @@ def new
     @results_validator.validate(@competition.id)
   end
 
+  def duplicate_checker
+    @competition = competition_from_params
+  end
+
+  def last_duplicate_checker_job
+    competition = competition_from_params
+
+    render status: :ok, json: competition.duplicate_checker_jobs.order(created_at: :desc).first
+  end
+
+  def compute_potential_duplicates
+    competition = competition_from_params
+
+    job = DuplicateCheckerJob.create!(
+      competition_id: competition.id,
+      status: DuplicateCheckerJob.statuses[:not_started],
+    )
+    ComputePotentialDuplicates.perform_later(job)
```

Comment:
The `ComputePotentialDuplicates` job should create its `DuplicateCheckerJob` in its own job body, instead of the record being passed in from the outside

Reply:
With the bigger schema migrations in place, I feel less strongly about this now. But you should definitely rename the variable from `job` to something like `job_run` or `run_record` or something. To make clear that the thing you're passing in is not the job itself.

Reply:
Okay, renamed to `job_run`.